### PR TITLE
Improve ``sizeof`` for ``pd.MultiIndex``

### PR DESCRIPTION
**Pull request by phofl**
- [ ] Closes #xxxx
- [ ] Tests added / passed
- [x] Passes `pre-commit run --all-files`

This is significantly closer to ``sys.getsizeof`` when the MultiIndex gets larger. Every MultiIndex level is a new Index itself.

cc @crusaderky

### CODE REVIEW

**Review by j-bennet — APPROVED** [review]
LGTM. I wonder what that magical number 400 is, in `sizeof_pandas_index`.

**Comment by phofl** [comment]
You are not alone. My guess is that it was necessary to make the test pass? No clue, the new implementation is significantly more accurate when the Index gets larger

File: dask/sizeof.py [hunk 1]
```diff
@@ -204,7 +204,7 @@ def sizeof_pandas_index(i):
 
     @sizeof.register(pd.MultiIndex)
     def sizeof_pandas_multiindex(i):
-        p = 400 + object_size(*i.levels)
+        p = sum(sizeof_pandas_index(lev) for lev in i.levels)
```
Comment (jrbourbeau) [inline dask/sizeof.py:207]: Minor suggestion

```suggestion
        p = sum(sizeof(lev) for lev in i.levels)
```

**Review by jrbourbeau — COMMENTED** [review]
> I wonder what that magical number 400 is, in sizeof_pandas_index

From looking through https://github.com/dask/dask/pull/9776, where this was added, IIUC this is to account for the sizeof an empty `pd.Index` (we then add the sizeof the data here to get the full `sizeof` estimate). 

```python
In [1]: from dask.sizeof import sizeof

In [2]: import pandas as pd

In [3]: s = pd.Series([1, 2, 3], index=[4, 5, 6])

In [4]: s.index[:0]
Out[4]: Index([], dtype='int64')

In [5]: sizeof(s.index[:0])
Out[5]: 400
```

@crusaderky is that correct?

If that is the case (again, not totally sure) I'm curious why we're hardcoding `400` instead of computing `sizeof` on a corresponding empty index since there are cases where `400` isn't always the correct size

```python
In [21]: s = pd.Series([1, 2, 3])

In [22]: sizeof(s.index[:0])
Out[22]: 524
```

Maybe this is for performance reasons?

**Comment by phofl** [comment]
Explanation:

The current implementation assumes that every level has object dtype, which is rarely correct 
The new implementation provides an accurate results for every non-object level and estimates object levels, which is better than just assuming every level is object
So in all-object cases it’s the same result while it should be closer to the truth for every other case

**Review by jrbourbeau — APPROVED** [review]

**Pull request merged**